src/StatusQ/Core/StatusAssetSettings.qml

```diff
@@ -4,22 +4,35 @@ import StatusQ.Core 0.1
 QtObject {
     id: root
 
+    //icon
     property string name
+    property url source
     property real width
     property real height
     property color color
     property color disabledColor
-    property url source
     property int rotation
     property bool isLetterIdenticon
     property int letterSize: emoji ? 11 : (charactersLen == 1 ? _oneLetterSize : _twoLettersSize)
     property int charactersLen: 1
     property string emoji
     property string emojiSize: _emojiSize
-    property StatusIconBackgroundSettings background: StatusIconBackgroundSettings {}
 
     // can't have QtObject { id: d } inside QtObject, using '_' to indicate private
     readonly property int _oneLetterSize: Math.max(15, root.width / 2.0)
     readonly property int _twoLettersSize: Math.max(12, root.width / 2.2)
     readonly property string _emojiSize: "%1x%1".arg(Math.max(16, root.width * 0.6))
+
+    //icon bg
+    property real bgWidth
+    property real bgHeight
+    property int bgRadius
+    property color bgColor: "transparent"
+
+    //image
+    property real imgWidth
+    property real imgHeight
+    property url imgSource
```

Comment:
Just one question here: how is `imgWidth` different from (icon) `width`? Same for `imhHeight` and `imgSource`; are there valid cases when both an image and icon are used together (sounds weird)

Reply:
yeah, in very rare cases (I think I saw 1 or 2) there image and icon together

Reply:
Ah hmmm, yeah inside a Loader which decides whether to use an icon or an image?

Reply:
Need to check if those 2 are used simultaneously, can't remember now.

Reply:
@caybro Indeed, it's either this or that, however some times there are minor differences when we load an image or an icon, mostly in width and height, which are manageable but I would like a better way of detecting this than what I used here as a quick solution. To give more insights: depending on if we have icon name set we load StatusIcon or if we have image StatusRoundedImage. It's either this or that. As a quick solution I'm using `root.asset.name.indexOf(".") === -1 ? roundedIcon : roundedImage` to load the correct component, however I don't like this, would you have any better idea?